tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/macro/MacroGeneratorIntegrationTest.xtend

```diff
@@ -0,0 +1,282 @@
+package org.testeditor.tcl.dsl.jvmmodel.macro
+
+import org.junit.Before
+import org.junit.Test
+import org.testeditor.dsl.common.testing.DummyFixture
+
+class MacroGeneratorIntegrationTest extends org.testeditor.tcl.dsl.jvmmodel.AbstractTclGeneratorIntegrationTest {
+
+	@Before
+	def void setup() {
+		parseAml(DummyFixture.amlModel)
+		parseTcl('''
+			package com.example
+			
+			# MyMacroCollection
+			
+			## EmptyMacro
+				template = "Do nothing"
+			
+			## EmptyNestedMacro
+				template = "Do nothing nested"
+				Macro: MyMacroCollection
+				- Do nothing
+			
+			## EmptyMacroWithUnusedParameter
+				template = "Do nothing with" ${unused}
+			
+			## ReadMacro
+				template = "Read some values"
+				Component: GreetingApplication
+				- value = Read value from <bar>
+			
+			## WriteMacro
+				template = "Set input to" ${value}
+				Component: GreetingApplication
+				- Set value of <Input> to @value
+			
+			## WaitMacro
+				template = "Wait for" ${x} "seconds"
+				Component: GreetingApplication
+				- Wait for @x seconds
+			
+			## SetValueAndWait
+				template = "Read and write value and wait" ${seconds} "seconds"
+				Component: GreetingApplication
+				- value = Read value from <bar>
+				Macro: MyMacroCollection
+				- Set input to @value
+				Macro: MyMacroCollection
+				- Wait for @seconds seconds
+		''')
+	}
+
+	@Test
+	def void emptyMacroWithoutParameter() {
+		// given
+		val tcl = '''
+			Macro: MyMacroCollection
+			- Do nothing
+		'''
+
+		// when
+		val generatedCode = tcl.parseAndGenerate
+
+		// then
+		generatedCode => [
+			assertContains('''
+				@Test
+				public void execute() throws Exception {
+				  
+				  reporter.enter(TestRunReporter.SemanticUnit.SPECIFICATION_STEP, "step1");
+				  
+				  // Macro: MyMacroCollection
+				  // - Do nothing
+				  macro_MyMacroCollection_EmptyMacro();
+				}
+			'''.indent(1))
+
+			assertContains('''
+				private void macro_MyMacroCollection_EmptyMacro() throws Exception {
+				  
+				}
+			'''.indent(1))
+		]
+	}
+
+	@Test
+	def void emptyMacroWithUnusedParameter() {
+		// given
+		val tcl = '''
+			Macro: MyMacroCollection
+			- Do nothing with "x"
+		'''
+
+		// when
+		val generatedCode = tcl.parseAndGenerate
+
+		// then
+		generatedCode => [
+			assertContains('''
+				// Macro: MyMacroCollection
+				// - Do nothing with "x"
+				macro_MyMacroCollection_EmptyMacroWithUnusedParameter("x");
+			'''.indent(2))
+
+			assertContains('''
+				private void macro_MyMacroCollection_EmptyMacroWithUnusedParameter(final String unused) throws Exception {
```

Comment:
I picked that as a default type as it handles passing values to an unused parameter more gracefully. If you pass "123" to an unused parameter of type Object you will get compile errors. There should be a warning if a parameter is not used, but those were explicitly not included in this PR to keep it "small".